### PR TITLE
[EDX-418] - Restructure authentication

### PR DESCRIPTION
**Pull request by tbedford**
## Description

Restructure authentication page into multiple topics.

IA hill epic:

* [EDX-326](https://ably.atlassian.net/browse/EDX-326)

This PR implements multiple tickets:

* [EDX-418](https://ably.atlassian.net/browse/EDX-418)
* [EDX-419](https://ably.atlassian.net/browse/EDX-419)
* [EDX-420](https://ably.atlassian.net/browse/EDX-420)
* [EDX-421](https://ably.atlassian.net/browse/EDX-421)

Includes TOC update:

image

The "auth & security" topic is removed from Key Concepts section. A new section "Authentication" has been added.

## Notes

* I think the token topic could be split into Ably tokens and JWTs. This would make the material more manageable.
* I split off "identified clients" and "token revocation" into their own topics, to try and make the "capabilities" and "token" topics respectively more manageable.
* Links have been updated.

## TODO

- [x] add caveat boxes 
- [x] add foldable boxes
- [x] check sub-heading against Miro
- [x] check wording and content
- [x] check links

## Review

* [Authentication](https://ably-docs-edx-418-auth--tmxnhp.herokuapp.com/docs/auth/)

### CODE REVIEW

File: content/auth/basic.textile [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+* The secret is passed directly by the client to Ably, so it is only permitted for connections that are over TLS, to prevent the key secret being intercepted.
+* All of the configured "capabilities":/auth/capabilities of the key are implicitly possible in any request, and clients that legitimately obtain this key may then abuse the rights for that key.
+* A client that authenticates using an API key can claim any client ID it chooses. Therefore this client ID cannot be trusted to represent the genuine identity of the client. Client IDs should be assigned by the server, once the client's credentials have been authenticated.
+
+<!-- CAVEAT-INPUT-->
+
+When selecting an Ably SDK for implementing basic authentication with Ably, you don't need to use the realtime interface. 
+
+As basic authentication is primarily designed for authenticating a secure server, in this situation, and assuming the server is only used for authentication, it is more efficient to use the REST interface, as the overhead associated with maintaining a realtime connection, is not required.
```
Comment (Sapples) [inline content/auth/basic.textile:112]: "associated with maintaining a realtime connection, is not required."

Remove comma at the end of the sentence. Might be worth a bit of a rewrite of the entire sentence as it's on the edge of being confusing with all the commas.

File: content/auth/capabilities.textile [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+The @channel-metadata@ permission works both ways. When associated with a specific channel or set of channels it allows you to use the "Channel Status API":/rest/channel-status#channel-status to request the status of that channel. When associated with the wildcard resource @'*'@ it takes on an additional meaning: as well as allowing channel status requests for all channels, it also allows you to "enumerate all active channels":/rest/channel-status#enumeration-rest. 
+
+<!-- CAVEAT-INPUT: Channel mode flags -->
+"Channel mode flags":LINKME enable a client to specify a subset of the capabilities granted by their token or API key as channel options. 
+
+Channel mode flags offer the ability for clients to use different capabilities for different channels, however, as they are flags and not permissions, they cannot be enforced by an authentication server. Channel mode flags also enable clients to be present on a channel without subscribing to presence events.
+
+h2(#capabilities-key). API key capabilities
+
+An "Ably API key":/auth#api-key can have single set of permissions, applied to any number of "channels":/LINKME or "queues":/general/queues.
```
Comment (Sapples) [inline content/auth/capabilities.textile:58]: ""Ably API key":/auth#api-key can have single set of permissions,"

can have **a** single set

File: content/auth/capabilities.textile [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+You can also choose whether to restrict the API key to only channels, only "queues":/general/queues, or to match a set of channel or queue names. If you've chosen to restrict the API key to *selected channels and queues*, you can use a comma separated list of resources the API key can access, making use of wildcard to provide access to areas of your app. It is worth noting an API key will provide the same permissions to all resources it has access to.
```
Comment (Sapples) [inline content/auth/capabilities.textile:60]: "making use of wildcard to provide"

 making use of wildcard**s** to provide

File: content/auth/index.textile [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+An Ably API key string has the following format: @I2E_JQ.OqUdfg:EVKVTCBlzLBPYJiCZTsIW_pqylJ9WVRB5K9P19Ap1y0@.  
+
+The API key is made up of three parts:
+
+1. @I2E_JQ@ is the public app ID (the part before the first period)
+2. @OqUdfg@ is the public app key ID (the part after the period and before the colon). @I2E_JQ.OqUdfg@ is the public API key ID (both the public app ID and app key ID together)
+3. @EVKVTCBlzLBPYJiCZTsIW_pqylJ9WVRB5K9P19Ap1y0@ is the API key secret and should never be shared with untrusted parties (the part after the colon)
+
+The API key string that is your private secret and should never be made public. This API key string is used in all of our SDKs and for authentication with the REST API.
```
Comment (Sapples) [inline content/auth/index.textile:26]: "The API key string that is your private secret and should never be made public."

The API key string is your private secret and should never be made public.

**Review by Sapples — CHANGES_REQUESTED** [review]
Couple of tiny typos, all good otherwise. Ran locally and links spot-checked, all tried links worked.

File: content/auth/basic.textile (hunk 1 above)
Comment (m-hulbert) [reply content/auth/basic.textile]: Good point - reworded the whole paragraph.

File: content/auth/capabilities.textile (hunk 2 above)
Comment (m-hulbert) [reply content/auth/capabilities.textile]: Fixed thanks.

File: content/auth/capabilities.textile (hunk 3 above)
Comment (m-hulbert) [reply content/auth/capabilities.textile]: Fixed, thanks.

File: content/auth/index.textile (hunk 4 above)
Comment (m-hulbert) [reply content/auth/index.textile]: Reworded this to make it more legible - good spot.

**Review by Sapples — APPROVED** [review]
LGTM

**Pull request merged**